pennylane/hf/molecule.py

```diff
@@ -110,3 +110,89 @@ def __init__(
         self.nuclear_charges = [atomic_numbers[s] for s in self.symbols]
 
         self.n_electrons = sum(np.array(self.nuclear_charges)) - self.charge
+
+        self.mo_coefficients = None
+
+    def generate_atomic_orbital(self, index):
+        r"""Return a function that computes the value of a basis function at a given position.
+
+        Args:
+            index (int): index of the basis function
+
+        Returns:
+            function: function that computes the basis function
+
+        **Example**
+
+        >>> symbols  = ['H', 'H']
+        >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+        >>> mol = qml.hf.Molecule(symbols, geometry)
+        >>> ao = mol.generate_atomic_orbital(0)
+        >>> ao(0.0, 0.0, 0.0)
+        0.62824688
+        """
+        l = self.basis_set[index].l
+        alpha = self.basis_set[index].alpha
+        coeff = self.basis_set[index].coeff
+        r = self.basis_set[index].r
+
+        coeff = coeff * primitive_norm(l, alpha)
+        coeff = coeff * contracted_norm(l, alpha, coeff)
+
+        lx, ly, lz = l
+
+        def orbital(x, y, z):
+            r"""Computes the value of a basis function at a given position.
```

Comment:
Maybe it could be shortened?
```suggestion
            r"""Evaluates the basis function at a given position.
```